packages/runtime-dom/src/apiCustomElement.ts

```diff
@@ -277,14 +283,46 @@ export class VueElement extends BaseClass {
       this._applyStyles(styles)
 
       // initial render
-      this._update()
+      this._update(parentCE)
     }
 
     const asyncDef = (this._def as ComponentOptions).__asyncLoader
     if (asyncDef) {
-      asyncDef().then(def => resolve(def, true))
+      asyncDef().then(def => {
+        resolve(def, true)
+        if (this._childResolve.length > 0) {
+          this._childResolve.forEach(
+            (childResolveFn: (parent: typeof this) => void) => {
+              childResolveFn(this)
+            }
+          )
+        }
+      })
     } else {
-      resolve(this._def)
+      // When parentNode is a custom element rendered by an asynchronous component,
+      // it is collected and waits for the asynchronous
+      // completion before executing the resolve function.
+      const getParentAsyncCE = (el: typeof this): typeof this | null => {
+        if (
+          el._isCE &&
+          el._def &&
+          (el._def as ComponentOptions).__asyncLoader
+        ) {
+          return el
+        } else if (el.parentNode) {
+          return getParentAsyncCE(el.parentNode as typeof this)
+        } else {
+          return null
+        }
+      }
+      const parentNode = getParentAsyncCE(this)
+      if (parentNode) {
```

Comment:
I feel we should not defer resolve children if `!(parentNode._def as ComponentOptions).__asyncResolved`